rasa/shared/core/flows/flow.py

```diff
@@ -981,16 +981,54 @@ def from_str(label: Optional[Text]) -> "CollectInformationScope":
             raise NotImplementedError
 
 
+@dataclass
+class SlotRejection:
+    """A slot rejection."""
+
+    if_: str
+    """The condition that should be checked."""
+    utter: str
+    """The utterance that should be executed if the condition is met."""
+
+    @staticmethod
+    def from_dict(rejection_config: Dict[Text, Any]) -> SlotRejection:
+        """Used to read slot rejections from parsed YAML.
+
+        Args:
+            rejection_config: The parsed YAML as a dictionary.
+
+        Returns:
+            The parsed slot rejection.
+        """
+        return SlotRejection(
+            if_=rejection_config.get("if"),
+            utter=rejection_config.get("utter"),
```

Comment:
Please use the direct access like `rejection_config["if"]`, otherwise you are violating the typing. It's `str` not `Optional[str]`. And I think it makes sense that these are required fields.

Reply:
Addressed: https://github.com/RasaHQ/rasa/pull/12802/commits/79b1cd1224fa7df417000941b36c3619a8e12ee9